src/OpenSpaceToolkit/Astrodynamics/Trajectory/Segment.cpp

```diff
@@ -136,25 +138,8 @@ Array<flightManeuver> Segment::Solution::extractManeuvers(const Shared<const Fra
         return {};
     }
 
-    // Loop through dynamics to find Thruster dynamics
```

Comment:
I think its worth keeping this logic but generalizing it to any dynamics from the provided argument